src/coreclr/interpreter/compiler.cpp

```diff
@@ -2219,27 +2222,20 @@ bool InterpCompiler::EmitNamedIntrinsicCall(NamedIntrinsic ni, CORINFO_CLASS_HAN
     }
 }
 
-bool InterpCompiler::EmitCallIntrinsics(CORINFO_METHOD_HANDLE method, CORINFO_SIG_INFO sig)
+bool InterpCompiler::EmitCallIntrinsics(CORINFO_METHOD_HANDLE method, CORINFO_SIG_INFO sig, bool newObj)
 {
+    if (newObj)
+        return false;
+
     const char *className = NULL;
     const char *namespaceName = NULL;
     const char *methodName = m_compHnd->getMethodNameFromMetadata(method, &className, &namespaceName, NULL, 0);
 
     if (namespaceName && !strcmp(namespaceName, "System"))
     {
-        if (className && !strcmp(className, "Environment"))
+        if (className && !strcmp(className, "Object"))
         {
-            if (methodName && !strcmp(methodName, "FailFast"))
-            {
-                AddIns(INTOP_FAILFAST); // to be removed, not really an intrisic
-                m_pStackPointer--;
-                return true;
-            }
-        }
-        else if (className && !strcmp(className, "Object"))
-        {
-            // This is needed at this moment because we don't have support for interop
-            // with compiled code, but it might make sense in the future for this to remain
+            // This is no longer needed, but it might make sense in the future for this to remain
```

Comment:
There are *many* trivial methods like this Object .ctor. It does not make sense to try to "fix them" by treating them as intrinsics. The proper way to fix this is via optimizing interpreter JIT.

My guess is that Object .ctor is not actually the most perf sensitive one among all the trivial methods. It is always going to be coupled with GC allocation that is relatively expensive that will make the cost of the transition less problematic.